stock_manager/cli.py

```diff
@@ -0,0 +1,558 @@
+        return False
+    
+    print('[+] Removing User From Databases...')
+    
+    DBUtils().update_users_database(DatabaseUpdateType.REMOVE, args.values[0])
+    
+    print('[*] Successfully Removed', args.values[0], 'From Databases')
+    return False
+
+
+def _list_items(search_value='') -> None:
+    """
+    Prints a formatted report of all stock items, optionally filtered by a search value.
+    
+    :param search_value: Optional substring to filter items.
+    """
+    
+    from stock_manager import DBUtils, StockStatus
+    
+    print('[+]', 'Gathering Item Data...' if not search_value else f'Searching For Items With "{search_value}"...')
+    widths = [3, 10, 12, 5, 10, 10, 8, 8, 6, 12, 20]
+    sum_widths = sum(widths)
+    all_data: list[dict[str, int | str | None]] = DBUtils().get_all_data_gs()
+    headers = [
+        'ID', 'Name', 'Manufacturer', 'Total',
+        'B750 Stock', 'B757 Stock', 'B750 Min',
+        'B757 Min', 'Excess', 'Status', 'Description'
+    ]
+    border = '=' * (sum_widths + 2 * len(widths))
+    
+    def truncate(text: str, width: int) -> str:
+        return text if len(text) <= width else f'{text[:width - 3]}...'
+    
+    def format_row(row_items: list[str]):
+        return '  '.join(
+                truncate(
+                        str(item), w
+                ).ljust(w)
+                    for item, w in zip(row_items, widths)
+        )
+    
+    print('\n[+] Stock Items Report')
+    print(border)
+    print(format_row(headers))
+    print(format_row(['-' * widths[i] for i in range(len(widths))]))
+    
+    i: int
+    data: dict[str, int | str | None]
+    out_of_stock = low_stock = in_stock = other = total = 0
+    for i, data in enumerate(all_data):
+        if search_value:
+            found = False
+            for value in data.values():
+                value = str(value)
+                if search_value.lower() in value.lower():
+                    found = True
+                    break
+            if not found:
+                continue
+        
+        row = [
+            i + 1,
+            data['Part #'],
+            data['Manufacturer'],
+            data['Total'],
+            data['B750'],
+            data['B757'],
+            data['Minimum'],
+            data['B750 Minimum'],
+            data['Excess'],
+            data['Stock Status'],
+            data['Description']
+        ]
+        match data['Stock Status']:
+            case StockStatus.OUT_OF_STOCK.value:
+                out_of_stock += 1
+            case StockStatus.LOW_STOCK.value:
+                low_stock += 1
+            case StockStatus.IN_STOCK.value:
+                in_stock += 1
+            case _:
+                other += 1
+        total += 1
+        print(format_row(row))
+    
+    print(border)
+    print(
+            '[*] Total Items:', total, '| Out Of Stock:', out_of_stock,
+            '| Low Stock:', low_stock, '| In Stock:', in_stock, '| Other:', other
+    )
```

Comment:
You put a lot of work into rolling your own ascii table here, but we've used `prettytable` quite extensively and been happy with it

https://pypi.org/project/prettytable/

Reply:
The experience of rolling your own will always be good practice haha, sorry we didn't mention it earlier